mods/ra/maps/evacuation/evacuation.lua

```diff
@@ -0,0 +1,365 @@
+		if actor.Owner == allies1 then
+			ReassignActors(TownUnits, neutral, allies1)
+			Utils.Do(TownUnits, function(a) a.Stance = "Defend" end)
+			allies1.MarkCompletedObjective(objFindEinstein)
+			objEinsteinSurvival = allies1.AddPrimaryObjective("Keep Einstein alive at all costs.")
+			Trigger.OnKilled(Einstein, function()
+				allies1.MarkFailedObjective(objEinsteinSurvival)
+			end)
+			Trigger.RemoveProximityTrigger(trigger)
+			SovietTownAttack()
+		end
+	end)
+
+	Trigger.OnEnteredProximityTrigger(YakAttackPoint.CenterPosition, WDist.FromCells(5), function(actor, trigger)
+		if not (Yak == nil or Yak.IsDead) or actor.Owner ~= allies1 then
+			return
+		end
+
+		Yak = Reinforcements.Reinforce(soviets, { "yak" }, { YakEntryPoint.Location, YakAttackPoint.Location + CVec.New(0, -10) }, 0, YakAttack)[1]
+	end)
+
+	Trigger.AfterDelay(ParabombDelay, SendParabombs)
+	Trigger.AfterDelay(ParatroopersDelay, SendParatroopers)
+	Trigger.AfterDelay(ReinforcementsDelay, SpawnAlliedReinforcements)
+end
+
+SpawnTanya = function()
+	Tanya = Actor.Create(TanyaType, true, { Owner = allies1, Location = TanyaLocation.Location })
+	Tanya.Stance = TanyaStance
+
+	if Map.Difficulty ~= "Easy" and allies1.IsLocalPlayer then
+		Trigger.AfterDelay(DateTime.Seconds(2), function()
+			Media.DisplayMessage("According to the rules of engagement I need your explicit orders to fire, Commander!", "Tanya")
+		end)
+	end
+end
+
+ReassignActors = function(actors, from, to)
+	for i, a in ipairs(actors) do
+		if a.Owner == from then
+			a.Owner = to
+		end
+	end
+end
+
+WorldLoaded = function()
+	neutral = Player.GetPlayer("Neutral")
+
+	-- Allies is the pre-set owner of units that get assigned to either the second player, if any, or the first player otherwise.
+	allies = Player.GetPlayer("Allies")
+
+	-- Allies1 is the player starting on the right, controlling Tanya
+	allies1 = Player.GetPlayer("Allies1")
+
+	-- Allies2 is the player starting on the left, defending the base
+	allies2 = Player.GetPlayer("Allies2")
+
+	soviets = Player.GetPlayer("Soviets")
+
+	for i, player in ipairs({ allies1, allies2 }) do
+		if player ~= nil and player.IsLocalPlayer then
+			Trigger.OnObjectiveAdded(player, function(p, id)
+				Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+			end)
+
+			Trigger.OnObjectiveCompleted(player, function(p, id)
+				Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+			end)
+
+			Trigger.OnObjectiveFailed(player, function(p, id)
+				Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+			end)
+
+			Trigger.OnPlayerWon(player, function()
+				Media.PlaySpeechNotification(player, "MissionAccomplished")
+			end)
+
+			Trigger.OnPlayerLost(player, function()
+				Media.PlaySpeechNotification(player, "MissionFailed")
+			end)
+		end
+	end
+
+	if allies2 == nil or allies2.IsLocalPlayer then
```

Comment:
We usually don't explicitly check for `nil`. I'd just use `if not allies2 or allies2.IsLocalPlayer then`.